Rebase of 10.10.x-8.3 onto v10.10.5

ydirson opened the pull request:
This branch regroups logically-related commits in thematic branches, and commits that would otherwise be alone in a `small-patches` branch.

stormi committed:
Change checkbox text for the EXT vs LVM option

Signed-off-by: Samuel Verschelde <stormi@laposte.net>

Orig-commit: b4ebf8174694799b19eaa5a946ba7f00827eeef5
Orig-commit: 03d3a84ce27976a1d9d6ea9e42a1e7492967f3e6

stormi committed:
Add software RAID support.

Originally-by: Nicolas Raynaud <github@nraynaud.com>

Also adds support for it in the answerfile.

the general form is :
<raid device="md127">
  <disk>sda</disk>
  <disk>sdb</disk>
</raid>

avoid question and display waiting message when running mdadm

Signed-off-by: Samuel Verschelde <stormi@laposte.net>

When restoring from backup on a software RAID install, install grub on
raid members instead of disk (which is a mdadm device)

Signed-off-by: BenjiReis <benjamin.reis@vates.fr>

Orig-commit: 6e168b43e0fd31cf3de9f578def1e10ba72ced35
Orig-commit: 7557a0077a65d9353a56d7a21d054ceadc29b796
Orig-commit: ee404b10b9809279600e0be121f88a675c0e2ef4
Orig-commit: e6497d1c602691104a3d750692ffceedb25c1221
Orig-commit: d33fe10fb1c36ee0326e48dbb8059a5fe42fa80b

stormi committed:
Add a netinstall boot option

The presence of this parameter changes the installer behavior,
preventing local source installation and pre-populating the xcp-ng
install url.

Originally-by: Nicolas Raynaud <github@nraynaud.com>

Use the `--netinstall` parameter that is automatically
generated at preinit stage and translate it into
`answers['netinstall'] = True`.

Signed-off-by: Samuel Verschelde <stormi@laposte.net>

Orig-commit: 36395c3c68e0d1afdd94ed8ec57d152640dde8b4
Orig-commit: 098a2f404c2d69bdd805d175943c993b88c2b4de
Orig-commit: 0b298cd20722ca0bae36ae7079df2b94491a25db

stormi committed:
Import XCP-ng GPG key to yum and rpm post-install

Signed-off-by: Samuel Verschelde <stormi@laposte.net>

Orig-commit: 9f0f094409a7f0c19a686cad3ed16f055cbbf155
Orig-commit: eb796dc6a82a0defdcbba1a0833a3356060c5396

fetzerms commented:
If we want the answer file to be kept/not overwritten, wouldn't it be the proper way to use answerfile_script (answerfile_generator) instead of answerfile_address? I do not think its a good idea to generally change the semantics of a fetch method, as its more a "fetch if not exists" than a simple fetch.

bdonnahue commented:
@fetzerms the answerfile_generator is absolutely the right approach. I will add some detail to #4 as well as the parameters.txt file re. how it works.

stormi committed:
Include mmcblk-devices in disk selection

Originally-by: Jöran Malek <alivedevil@users.noreply.github.com>
Signed-off-by: Samuel Verschelde <stormi@laposte.net>

Orig-commit: fc1b4cf120c68cc03ba7a82251a51ba6d9b7533d

stormi committed:
Use xcp-ng-deps instead of groups.xml

Signed-off-by: Samuel Verschelde <stormi@laposte.net>

Orig-commit: 2731116cba1ef80a2628c041002f82258e1be5cb

stormi committed:
kernel-alt support

- show kernel-alt warning

Originally-by: Rushikesh Jadhav <rushikesh7@gmail.com>

- Update warning message when installing with kernel-alt
- cleaner reboot action on that screen: return EXIT
- fix detection of the kernel-alt boot parameter of the installer
- install kernel-alt when booting on install-alt
- create initrd and update grub configuration as post-install task

Signed-off-by: Samuel Verschelde <stormi@laposte.net>

Orig-commit: 243ce81f0ff4ad17c99c1ca6af5f3326af1c94db
Orig-commit: d9a8a3eb609b6c0493fb3b6db03fc91bc5321f45

benjamreis committed:
feat: keep 'sched-gran' grub entry on upgrade (#21)

If Xen's scheduling mode has been defined previously, we keep the value when upgrading.

Signed-off-by: BenjiReis <benjamin.reis@vates.fr>

Orig-commit: f0479a7180318493852314059ca017d4c24f0d9c

ydirson committed:
Add support for globally disabling repo_gpgcheck

While developing and/or installing from an already-verified ISO, yum's
repo_gpgcheck just gets in the way.  This provides a general kill-switch
for checking repomd.xml.asc, through commandline or answerfile.

Signed-off-by: Yann Dirson <yann.dirson@vates.fr>

ydirson committed:
Add support for globally disabling gpgcheck

Similar to no-repo-gpgcheck but for RPM sigs.

Signed-off-by: Yann Dirson <yann.dirson@vates.fr>

ydirson committed:
answerfile: support *gpgcheck override for each <source>

Global *gpgcheck flag can change the default from True to False, and these
new flags allow to override this default on a per-source basis.

Signed-off-by: Yann Dirson <yann.dirson@vates.fr>

ydirson committed:
Remove mention of Technical Support Representative in error messages

Originally-by: Samuel Verschelde <stormi@laposte.net>
Signed-off-by: Yann Dirson <yann.dirson@vates.fr>

Orig-commit: b20e3033ecc1072f01f1b4cf619f3b2b952e592f

ydirson committed:
Merge branch '8.2.1-small-patches'

Rebase of simple 8.2.1 patches onto 8.3 codebase

stormi committed:
Merge pull request #2 from xcp-ng/yann/gpgcheck

Global and per-source gpgcheck flags

ydirson committed:
Fix interactive install to honor no-*gpgcheck from commandline

When originally implementing the per-source gpgcheck flags in
answerfile[1], the full code was moved to an anwerfile-only location,
breaking the original interactive-install implementation (and then
following code review[2] the working code for interactive install
disappeared further from the patch series).

This moves the Repository flag-setting to `add_repos()` common code,
while leaving the flag computation to the caller, since only the
answerfile case has to do any non-trivial logic.

- [1] https://github.com/xcp-ng/host-installer/commit/06b700789f2c22d1c78285f9583937511e4d3c07
- [2] https://github.com/xcp-ng/host-installer/pull/2#discussion_r1012080594

Signed-off-by: Yann Dirson <yann.dirson@vates.fr>

stormi committed:
Update URL for netinstall in 8.3

Signed-off-by: Samuel Verschelde <stormi-xcp@ylix.fr>

ydirson committed:
RAID support: zap partition tables of disks before building a RAID

Failing to do so can leave the kernel's idea of the partitions in weird
states.  There are likely bugs underlying those weird state, we just don't
want to be impacted by them at this point :)

Signed-off-by: Yann Dirson <yann.dirson@vates.fr>

ydirson committed:
Merge branch 'install-on-raid1'

ydirson committed:
Local SR: use a RadioBar for LVM/EXT choice

Signed-off-by: Yann Dirson <yann.dirson@vates.fr>

ydirson committed:
Merge branch 'lvm-vs-ext'

ydirson committed:
Merge branch 'netinstall'

benjamreis reviewed (approved):
LGTM
